extensions/ticktick/src/search.tsx

```diff
@@ -29,17 +54,54 @@ const TickTickSearch: React.FC<Record<string, never>> = () => {
   const { searchTasks, isSearching } = useSearchTasks({ searchQuery, isInitCompleted });
 
   const isLoading = useMemo(() => {
-    if (!isInitCompleted) {
+    if (!isInitCompleted || !isLocalDataLoaded || !isAllTasksLoaded) {
       return true;
     }
     return isSearching;
-  }, [isInitCompleted, isSearching]);
+  }, [isAllTasksLoaded, isInitCompleted, isLocalDataLoaded, isSearching]);
 
   const onFilterChange = (filterId: string) => {
     setProjectFilter(filterId);
+    LocalStorage.setItem("searchProjectFilter", filterId);
   };
 
+  useEffect(() => {
+    const getTasks = async () => {
+      setIsAllTasksLoaded(false);
+      const tasks = await getTasksByProjectId(projectFilter);
+      setSections(tasks);
+      setTimeout(() => {
+        setIsAllTasksLoaded(true);
+      });
+    };
+    if (isLocalDataLoaded) {
+      getTasks();
+    }
+  }, [isLocalDataLoaded, projectFilter]);
```

Comment:
Note that you can use Raycast's hooks in `@raycast/utils` like `useFetch`, `usePromise` or `useCachedPromise`. This would make it easier to manage the loading/error states.